add CacheFlow class, improve cache implementation (#30)
datahq/dataflows: PROCESSORS.md

- `updated_id_column` - Optional name of a column that will be added to the output data containing the id of the updated row in DB.

#### checkpoint

Save results from running a series of steps, if checkpoint exists - loads from checkpoint instead of running the steps.

> Checkpoint invalidation should be handled manually - by deleting the checkpoint path (by default at `.checkpoints/checkpoint_name`)

Reviewer: ```suggestion
Checkpoint invalidation should be handled manually - by deleting the checkpoint path (by default at `.checkpoints/<checkpoint_name>`)
```

```python
def checkpoint(checkpoint_name, checkpoint_path='.checkpoints', steps=None, resources=None):
    pass
```

> - `checkpoint_name` - The checkpoint is saved to a datapackage at `cache_path/checkpoint_name`

Reviewer: ```suggestion
- `checkpoint_name` - The checkpoint is saved to a datapackage at `cache_path/<checkpoint_name>`
```

- `cache_path` - Relative or absolute path to save checkpoints at, default to relative path `.checkpoints`
- `steps` - Iterable of steps to checkpoint, when not provided, uses the previous steps in the flow
- `resources` - Limit the checkpointing only to specific resources, same semantics as `load` processor `resources` argument

> To improve code readiblity, the checkpoint processor can be place anywhere in the flow and it will checkpoint the previous steps:

Reviewer: ```suggestion
To improve code readability, the checkpoint processor can be placed anywhere in the flow and it will checkpoint the previous steps:
```
